Common/PM_datatypes.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+// please change this name
+enum WaypointType {PATH_FOLLOW = 0, ORBIT_FOLLOW, HOVER_WAYPOINT, TAKEOFF_WAYPOINT, LANDING_WAYPOINT, TRANSITION_WAYPOINT, TELEOP_MODE};
+
+struct CommandsFromSMToPM{
+    // please rename this
+    WaypointType waypoint_type;
+    WaypointsCommand telemetry_commands;
+    JetsonToZpMovementCommand jetson_commands; 
+    LandingInitiationCommand landing_initiation; 
+    LosSFData sf_data;
+};
+
+struct JetsonToZpMovementCommand {
+  float x;
+  float y;
+  float z;
+  float heading;
+};
+
+struct LandingInitiationCommand {
+    bool start_landing; 
+};
+
+struct JetsonMovementRequest {
+    bool request;
+}
+
+struct JetsonOdometryData {
+    LosSFData sensorData;
+}
+
```

Comment:
Not sure if this would go in AM or PM datatypes (i think PM since AM's general receiving struct might be different?)
```
typedef struct CommandsForAM_t{
  WaypointType waypoint_type;  // not sure if still needed, added just in case

  // heading unit vector and magnitude
  float dist_x; 
  float dist_y; 
  float dist_z; 
  float magnitude; // Magnitude distance to waypoint target
  float heading; // heading at target waypoint
  double speed_target; // Target velocity of drone approaching target
} CommandsForAM;
```

Reply:
Perhaps we can come up with a good naming convention that includes the source and destination. A lot of the structs in this file make no mention of PM and would be confusing to reference from other managers